packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageViewWrapper.kt

```diff
@@ -420,10 +431,25 @@ class ExpoImageViewWrapper(context: Context, appContext: AppContext) : ExpoView(
       shouldRerender = false
       loadedSource = null
       transformationMatrixChanged = false
+      clearViewBeforeChangingSource = false
       return
     }
 
     if (sourceToLoad != loadedSource || shouldRerender || (sourceToLoad == null && placeholder != null)) {
+      if (clearViewBeforeChangingSource) {
+        val activeView = if (firstView.drawable != null) {
+          firstView
+        } else {
+          secondView
+        }
+
+        activeView
+          .recycleView()
+          ?.apply {
+          clear(requestManager)
+        }
```

Comment:
is that correct formatting?
```suggestion
          ?.apply {
            clear(requestManager)
          }
```